internal/controller/pkg/resolver/reconciler.go

```diff
@@ -554,7 +559,12 @@ func (r *Reconciler) findDependencyVersionToUpgrade(ctx context.Context, ref nam
 	return "", errors.Errorf(errFmtNoValidVersion, dep.Identifier(), dep.GetParentConstraints())
 }
 
-func (r *Reconciler) getPackageWithID(ctx context.Context, id string, t v1beta1.PackageType) (v1.Package, error) { //nolint:gocognit // TODO(ezgidemirel): This function can be simplified by using a single lister for all package types.
+func (r *Reconciler) getPackageWithRef(ctx context.Context, pkgRef string, t v1beta1.PackageType) (v1.Package, error) { //nolint:gocognit // TODO(ezgidemirel): This function can be simplified by using a single lister for all package types.
+	id, _, err := splitPackage(pkgRef)
```

Comment:
Why this? Did we catch another unrelated issue and fixing it?

Reply:
yes, I found another default registry related bug while testing